ci-operator/step-registry/upi/install/aws/cluster/upi-install-aws-cluster-commands.sh

```diff
@@ -0,0 +1,349 @@
+# that oc for all actions.
+export PATH=/tmp:${PATH}
+GATHER_BOOTSTRAP_ARGS=
+
+function gather_bootstrap_and_fail() {
+  if test -n "${GATHER_BOOTSTRAP_ARGS}"; then
+    openshift-install --dir=${ARTIFACT_DIR}/installer gather bootstrap --key "${SSH_PRIV_KEY_PATH}" ${GATHER_BOOTSTRAP_ARGS}
+  fi
+
+  return 1
+}
+# ensure LEASED_RESOURCE is set
+if [[ -z "${LEASED_RESOURCE}" ]]; then
+  echo "Failed to acquire lease"
+  exit 1
+fi
+
+cp "$(command -v openshift-install)" /tmp
+mkdir ${ARTIFACT_DIR}/installer
+
+echo "Installing from initial release ${RELEASE_IMAGE_LATEST}"
+SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+PULL_SECRET_PATH=${CLUSTER_PROFILE_DIR}/pull-secret
+OPENSHIFT_INSTALL_INVOKER=openshift-internal-ci/${JOB_NAME}/${BUILD_ID}
+AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred
+EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+PULL_SECRET=${CLUSTER_PROFILE_DIR}/pull-secret
+OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="${RELEASE_IMAGE_LATEST}"
+
+export OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+export PULL_SECRET_PATH
+export OPENSHIFT_INSTALL_INVOKER
+export AWS_SHARED_CREDENTIALS_FILE
+export EXPIRATION_DATE
+export PULL_SECRET
+
+mkdir -p ~/.ssh
+cp "${SSH_PRIV_KEY_PATH}" ~/.ssh/
+cp ${SHARED_DIR}/install-config.yaml ${ARTIFACT_DIR}/installer/install-config.yaml
+export PATH=${HOME}/.local/bin:${PATH}
+
+pushd ${ARTIFACT_DIR}/installer
+
+base_domain=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["baseDomain"])')
+AWS_REGION=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["platform"]["aws"]["region"])')
+CLUSTER_NAME=$(python3 -c 'import yaml;data = yaml.full_load(open("install-config.yaml"));print(data["metadata"]["name"])')
+echo ${AWS_REGION} > ${SHARED_DIR}/AWS_REGION
+echo ${CLUSTER_NAME} > ${SHARED_DIR}/CLUSTER_NAME
+MACHINE_CIDR=10.0.0.0/16
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
+openshift-install --dir=${ARTIFACT_DIR}/installer create manifests
+sed -i '/^  channel:/d' ${ARTIFACT_DIR}/installer/manifests/cvo-overrides.yaml
+rm -f ${ARTIFACT_DIR}/installer/openshift/99_openshift-cluster-api_master-machines-*.yaml
+rm -f ${ARTIFACT_DIR}/installer/openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+sed -i "s;mastersSchedulable: true;mastersSchedulable: false;g" ${ARTIFACT_DIR}/installer/manifests/cluster-scheduler-02-config.yml
+
+echo "Creating ignition configs"
+openshift-install --dir=${ARTIFACT_DIR}/installer create ignition-configs &
+wait "$!"
+
+cp ${ARTIFACT_DIR}/installer/bootstrap.ign ${SHARED_DIR}
+BOOTSTRAP_URI="https://${JOB_NAME_SAFE}-bootstrap-exporter-${NAMESPACE}.svc.ci.openshift.org/bootstrap.ign"
+export BOOTSTRAP_URI
+
+# begin bootstrapping
+if openshift-install coreos print-stream-json 2>/tmp/err.txt >coreos.json; then
+  RHCOS_AMI="$(jq -r .architectures.x86_64.images.aws.regions[\"${AWS_REGION}\"].image coreos.json)"
```

Comment:
See https://bugzilla.redhat.com/show_bug.cgi?id=2047799. The preferred approach with jq is to pass in environment variables via the --arg flag.

Reply:
I think it was more of a quoting issue? This code does not have the single quotes but the code in templates does.
https://github.com/openshift/release/commit/3c29ef1ea4988141d8c97cd0db55e8dad554ba09#diff-ba2cb3cdc0f179f886cf0bf9167c1251e87209e78d18f8bff73cec1463afd02aR1314

Reply:
Yes, the issue in the template was quoting. But I do think it is better to use the --arg flag and avoid the quoting quagmire altogether. Just a nit: Feel free to keep what you have.
```suggestion
  RHCOS_AMI="$(jq -r --arg region "$AWS_REGION" '.architectures.x86_64.images.aws.regions[$region].image' coreos.json)"
```